ldcbench.data-generator/src/main/java/org/dice_research/ldcbench/benchmark/DataGenerator.java

```diff
@@ -203,30 +215,29 @@ public void handleRdfGraph(int senderId, GraphMetadata gm) {
         dataGeneratorsChannel.basicConsume(queueName, true, consumer);
     }
 
-    protected void addInterlinks(GraphBuilder g) {
+    protected void addInterlinks(GraphBuilder g, int numberOfnodesLinks) {
         int numberOfInternalNodes = g.getNumberOfNodes();
         Random random = new Random(seedGenerator.getNextSeed());
         for (Map.Entry<Integer, GraphMetadata> entry : rdfMetadata.entrySet()) {
             int targetNodeGraph = entry.getKey();
-//            GraphMetadata gm = entry.getValue();
-
-            // use random node
-            int nodeWithOutgoingLink = random.nextInt(numberOfInternalNodes);
-//            if (gm.entranceNodes.length == 0) {
-//                throw new IllegalStateException("Node " + nodeId + " needs to link to node " + targetNodeGraph
-//                        + " but there are no entrypoints.");
-//            }
-            // get node in target graph
-            int entranceInTargetGraph = 0;// FIXME use gm.entranceNodes[random.nextInt(gm.entranceNodes.length)];
-            // add a new node
-            int externalNode = g.addNode();
-            g.setGraphIdOfNode(externalNode, targetNodeGraph, entranceInTargetGraph);
-
-            // FIXME don't always use edge type 0
-            int propertyId = 0;
-            g.addEdge(nodeWithOutgoingLink, externalNode, propertyId);
-            LOGGER.debug("Added the edge ({}, {}, {}) where the target is node {} in graph {}.", nodeWithOutgoingLink,
-                    propertyId, externalNode, entranceInTargetGraph, targetNodeGraph);
+            GraphMetadata gm = entry.getValue();
+            if (gm.entranceNodes.length == 0) {
+                throw new IllegalStateException("Node " + getNodeId() + " needs to link to node " + targetNodeGraph
+                        + " but there are no entrypoints.");
+            }
+            for(int i = 0; i < numberOfnodesLinks; i++) {                
+                // get node in target graph
+                int entranceInTargetGraph = gm.entranceNodes[random.nextInt(gm.entranceNodes.length)];
+                // use random node
+                int nodeWithOutgoingLink = random.nextInt(numberOfInternalNodes);
+                // add a new node
+                int externalNode = g.addNode();
+                g.setGraphIdOfNode(externalNode, targetNodeGraph, entranceInTargetGraph);
+                int propertyId = i;
+                g.addEdge(nodeWithOutgoingLink, externalNode, propertyId);
+                LOGGER.debug("Added the edge ({}, {}, {}) where the target is node {} in graph {}.", nodeWithOutgoingLink,
+                        propertyId, externalNode, entranceInTargetGraph, targetNodeGraph);
+            }
```

Comment:
Although this implementation looks reasonable, it does not fulfill the requirements that ORCA has for the links between the nodes. Please keep in mind that the graph generation of ORCA has to guarantee that all graphs can be crawled. Depending on the generation that is used to create a graph, the graph may have more than 1 entrance node. However, this implementation links only `numberOfnodesLinks` of the target graph even if this graph may have more entrance nodes. This leads to the situation that it might not be possible to crawl the complete target graph.

Please have a look at this piece of code that already implements the creation of 1 link to each entrance nodes in the target graph: https://github.com/dice-group/orca/pull/80/files#diff-d227b2e8ca23cf5fec97ef332066f163ae8bfee1f2a07fe36476bf79de11f01fR207-R226
Your extension would simply be to ensure that instead of only 1 link, `numberOfnodesLinks` links are created for each entrance node in the target graph.